### DISCUSSION TITLE
How to fix ... TypeError: Cannot read properties of null (reading 'checked')

### DISCUSSION THREAD

**Discussion by Elindo586** [vercel/next.js#36948] [Help] [answered]
I think this is related to the problem that Nextjs can't read the html on the first load? 

The code link is here:
https://github.com/Elindo586/nextprojecta/blob/main/pages/components/pump-form.js

The error is this:
![error](https://user-images.githubusercontent.com/93617433/168612384-10714308-a3ca-4a51-aa67-38bf9163d614.PNG)

the code snip this this:
```
const PumpForm = () => {
  useEffect(() => {
    if (document.getElementById("pump1").checked) {
      document.getElementById("fieldsetpump2").style.display = "none";
      document.getElementById("fieldsetpump3").style.display = "none";
    }
    if (document.getElementById("p1fixedselect").checked) {
      document.getElementById("p1controlselect").style.display = "none";
    }
  }, []);

```
Later on in the return() section I go into selector inputs to run 3 different functions depending on the selection.. by default the first input is checked, but Nextjs is giving an error of null.... 

What can I do to fix it? 

```
    {/* <input type="radio" name="pumpselect" id= pump1 value="singlePump" onclick = "singlePumpf()" checked /> */}
            <input
              type="radio"
              name="pumpselect"
              id="pump1"
              value="singlePump"
              onClick={singlePumpf}
              defaultChecked
            />
            <label htmlFor="pump1"> Single Pump</label>
            <br />

            <input
              type="radio"
              name="pumpselect"
              id="pump2"
              value="doublePump"
              onClick={doublePumpf}
            />
            <label htmlFor="pump2"> Double Pump</label>
            <br />

            <input
              type="radio"
              name="pumpselect"
              id="pump3"
              value="Triplepump"
              onClick={triplePumpf}
            />
```

**Comment by icyJoseph** [comment, marked as answer]
Hi,

This is again where, using `refs` would be much better.

```ts
const PumpForm = () => {
  useEffect(() => {
    const pump1 = document.getElementById("pump1");
    if (pump1 && pump1.checked) {
      document.getElementById("fieldsetpump2").style.display = "none";
      document.getElementById("fieldsetpump3").style.display = "none";
    }
    if (document.getElementById("p1fixedselect").checked) {
      document.getElementById("p1controlselect").style.display = "none";
    }
  }, []);
```

You'd need to do a check like that, because `getElementById` can return `null`, as you can see. 

So you gotta make sure that it exists, before you try to read a property on it. 


### Edit

Your component returns nothing, so nothing is even being rendered at all.

```jsx
function p1Controlf() {
    if (document.getElmentById("p1variableselect").checked) {
      document.getElementById("p1controlselect").style.display = "block";
    }
    return  (
      <div className="col-md-12 bg-light">
        <p>
```

This is a syntax error, not a Next, nor React error.

You should close the scope of `p1Controlf`, and then return.

> I think this is related to the problem that Nextjs can't read the html on the first load?

No, it just the function that's returning nothing, so nothing is rendered at all.

### Edit 2

There's also Link tags but `Link` is not imported, and you pass `to` instead of `href`? Perhaps it comes from another framework?

Also, `import { useEffect, React } from 'react'` is not correct, and in Next.js you just need: `import { useEffect } from 'react';`


### Edit 3

Last but not least, you could just pass style objects to the `fieldsets`, or use classnames, etc., there's many ways in which you could avoid having to set inline styles like you have. However, first you gotta fix the way you are checking checked, and getting each element.

**Comment by Elindo586** [comment]
Oh wait.... I was missing the "function" part.

I am also cleaning some parts and commenting some areas of this code which I don't need.

These functions would run onClick, so I might not even need useEffect.

BTW.... this code went from plain html coding to REACT, and now to Next.js....  there is lots of cleaning and could have been a different approach if the original code was done straight in Next.js.. 

Moving on... I'll post later see if its a fix.

**Comment by Elindo586** [comment]
Thanks...  fixed...

Missing part of the function...  the useEffect wasn't needed.... and I imported the next/link and fixed the link tag..

**Comment by hritu8** [comment]
i want to implement null checks warnings when i execute npm run build
